department-of-veterans-affairs/va.gov-cms
# VACMS-15968: Remove Events CTA Feature Toggles

## Description
This PR removes 2 feature toggles that were temporarily added to control the display of the 'Add to National Outreach Calendar' checkbox.

Additionally, this PR:

- Makes use of Entity Bundle Classes and creates a new Event bundle class
- Moves pre-save logic from the EventSubscriber to the new Event bundle class
- Creates a new interface for encapsulating Event outreach related constants (and future methods if needed)
- Creates a new trait for core handling of national outreach calendar checkbox logic/form manipulations.
- Removes unnecessary and temporary methods now that we always want to show the checkbox for all Event editors.

Relates to #15968.

## Testing done
Updated the test to remove feature toggles.

The net change is that the 'add to nat' calendar' checkbox behaves the same way it does on prod today.

## Screenshots
![Screenshot 2023-11-27 at 11 16 39 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/c04fef15-ee1c-4593-83d9-492edd2df9c1)

## QA steps
As a content admin
1. Go to /node/add/event
   - [x] Validate that the 'Publish to the National Outreach Calendar' calendar checkbox is present
2. Check the checkbox, and fill in other Event details that are required.
3. Save the event.
   - [x] Validate that the Event detail page contains 'Additional Listings' with a link to 'Outreach events'

As a 'Outreach hub' only user (a user with only the 'Outreach hub' Section assigned, eg: tyana.johnson@va.gov)
4. Go to /node/add/event
   - [x] Validate the 'Where should the event be listed?' field is pre-populated with the 'Outreach and Events: Outreach events' option.
   - [x] Validate there is a checkbox below that is checked and,
   - [x] Validate that the label for the checkbox reads: 'This event will automatically be published to the National Outreach Calendar'
5.  Fill in Event require details and save the Event.
   - [x] Validate that the Event detail page does NOT contain 'Additional Listings'
   - [x] Validate that the Event detail page contains 'Where should the event be listed?' with a link to 'Outreach events' 

### Definition of Done


### Select Team for PR review
